### ISSUE TITLE
[BUG] Azure Identity 1.4.0 - authentication causes timeout

### ISSUE THREAD

**Issue by spinatelli** [Azure/azure-sdk-for-java#25084]
**Describe the bug**
Our API layer uses azure-identity and Managed Identity to perform actions on EventHubs and other resources. The Managed Identity is attached using the AAD pod identity, since we are running on AKS.

We upgraded from azure-identity 1.3.7 to 1.4.0 and since then we get timeouts on our APIs when trying to authenticate.

***Exception or Stack Trace***
```
2021-10-28 09:09:53.827  INFO 7 --- [or-http-epoll-3] c.a.identity.ManagedIdentityCredential   : Azure Identity => Managed Identity environment: AZURE VM IMDS ENDPOINT
2021-10-28 09:09:53.831  INFO 7 --- [or-http-epoll-3] c.a.identity.ManagedIdentityCredential   : Azure Identity => getToken() result for scopes [https://management.core.windows.net//.default]: SUCCESS
2021-10-28 09:09:53.832  INFO 7 --- [or-http-epoll-3] c.a.c.implementation.AccessTokenCache    : Acquired a new access token.
2021-10-28 09:10:23.894  INFO 7 --- [     parallel-1] c.a.identity.ManagedIdentityCredential   : Azure Identity => Managed Identity environment: AZURE VM IMDS ENDPOINT
2021-10-28 09:10:23.896  INFO 7 --- [     parallel-1] c.a.identity.ManagedIdentityCredential   : Azure Identity => getToken() result for scopes [https://management.core.windows.net//.default]: SUCCESS
2021-10-28 09:10:23.897  INFO 7 --- [     parallel-1] c.a.c.implementation.AccessTokenCache    : Acquired a new access token at 1635326822 seconds after expiry. Retry may be attempted after 30 seconds.
```

**To Reproduce**
Run AKS with AAD pod identity mapping for the container, run something similar to the code snippet. Works in 1.3.7, times out in 1.4.0

***Code Snippet***
Kotlin code we use, works in 1.3.7
```

 val credentials = ManagedIdentityCredentialBuilder()
        .clientId(<managedIdentityClientId>)
        .build()
val profile = AzureProfile(
        AzureConfiguration.getGlobalConfiguration().get(AzureConfiguration.PROPERTY_AZURE_TENANT_ID),
        AzureConfiguration.getGlobalConfiguration().get(AzureConfiguration.PROPERTY_AZURE_SUBSCRIPTION_ID),
        AzureEnvironment.AZURE
    )
val manager = EventHubsManager.authenticate(azureCredential, azureProfile)
```
**Expected behavior**

Logs from a working 1.3.7 library

```
2021-10-28 09:52:43.297  INFO 6 --- [or-http-epoll-3] c.a.identity.ManagedIdentityCredential   : Azure Identity => Managed Identity environment: AZURE VM IMDS ENDPOINT
2021-10-28 09:52:43.298  INFO 6 --- [or-http-epoll-3] c.a.identity.ManagedIdentityCredential   : Azure Identity => getToken() result for scopes [https://management.core.windows.net//.default]: SUCCESS
2021-10-28 09:52:43.300  INFO 6 --- [or-http-epoll-3] c.a.c.implementation.AccessTokenCache    : Acquired a new access token.
<OTHER APPLICATION LOGS>
```

**Screenshots**

**Setup (please complete the following information):**
 - Library/Libraries: azure-identity 1.4.0
 - Kotlin 1.5.31
 - App Server/Environment: AKS
 - Frameworks: Spring Boot

  
**Additional context**

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

**Comment by timja** [comment]
Hi 

We hit this as well MSI auth is completely broken in 1.4.0 and 1.4.1.
Some logs can be seen in https://github.com/jenkinsci/azure-vm-agents-plugin/issues/328

For now we've rolled back to 1.3.7

Happy to share code (it's all open source) if you can't reproduce in some way although I assume it's pretty easy.

**Comment by knutwannheden** [comment]
I think we've also run into this. Although 1.4.0 appears to work in our case while 1.4.1 is completely broken.

**Comment by timja** [comment]
The log messages are slightly different between 1.4.0 and 1.4.1 but both don't work.

1.4.0:
```
2021-11-19 20:01:24.366+0000 [id=293]	INFO	c.a.c.util.logging.ClientLogger#performLogging: Azure Identity => Managed Identity environment: AZURE VM IMDS ENDPOINT
2021-11-19 20:01:24.366+0000 [id=293]	INFO	c.a.c.util.logging.ClientLogger#performLogging: Azure Identity => getToken() result for scopes [https://management.core.windows.net//.default]: SUCCESS
2021-11-19 20:01:24.367+0000 [id=293]	INFO	c.a.c.util.logging.ClientLogger#info: Acquired a new access token at 1637265655 seconds after expiry. Retry may be attempted after 30 seconds.
2021-11-19 20:01:54.377+0000 [id=295]	INFO	c.a.c.util.logging.ClientLogger#performLogging: Azure Identity => Managed Identity environment: AZURE VM IMDS ENDPOINT
2021-11-19 20:01:54.377+0000 [id=295]	INFO	c.a.c.util.logging.ClientLogger#performLogging: Azure Identity => getToken() result for scopes [https://management.core.windows.net//.default]: SUCCESS
2021-11-19 20:01:54.377+0000 [id=295]	INFO	c.a.c.util.logging.ClientLogger#info: Acquired a new access token at 1637265714 seconds after expiry. Retry may be attempted after 30 seconds.
2021-11-19 20:02:24.387+0000 [id=298]	INFO	c.a.c.util.logging.ClientLogger#performLogging: Azure Identity => Managed Identity environment: AZURE VM IMDS ENDPOINT
```

1.4.1:
```
-11-19 20:13:17.194+0000 [id=16]	INFO	c.a.c.util.logging.ClientLogger#info: Package versions: jackson-annotations=2.13.0, jackson-core=2.13.0, jackson-databind=2.13.0, jackson-dataformat-xml=2.13.0, jackson-datatype-jsr310=2.13.0, azure-core=1.22.0, Troubleshooting version conflicts: https://aka.ms/azsdk/java/dependency/troubleshoot
2021-11-19 20:13:18.327+0000 [id=16]	SEVERE	c.a.c.util.logging.ClientLogger#performLogging: MSI: Failed to acquire tokens after retrying 3 times
2021-11-19 20:13:18.334+0000 [id=16]	SEVERE	c.a.c.util.logging.ClientLogger#performLogging: Azure Identity => ERROR in getToken() call for scopes [https://management.core.windows.net//.default]: MSI: Failed to acquire tokens after retrying 3 times
2021-11-19 20:13:18.334+0000 [id=16]	SEVERE	c.a.c.util.logging.ClientLogger#error: Failed to acquire a new access token.
2021-11-19 20:13:48.343+0000 [id=71]	SEVERE	c.a.c.util.logging.ClientLogger#performLogging: MSI: Failed to acquire tokens after retrying 3 times
2021-11-19 20:13:48.343+0000 [id=71]	SEVERE	c.a.c.util.logging.ClientLogger#performLogging: Azure Identity => ERROR in getToken() call for scopes [https://management.core.windows.net//.default]: MSI: Failed to acquire tokens after retrying 3 times
2021-11-19 20:13:48.343+0000 [id=71]	SEVERE	c.a.c.util.logging.ClientLogger#error: Failed to acquire a new access token.
```

**Comment by g2vinay** [comment]
Thank you for reporting this issue.
I am investigating it and will update here soon.

**Comment by g2vinay** [comment]
@spinatelli @timja @knutwannheden 

This issue should be fixed in 1.4.2 version of azure-identity.
You can upgrade to that version.
Let me know if you see any other problems and we can reopen this issue.
